src/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -81,7 +81,7 @@ internal static AuthorityInfo FromAuthorityUri(string authorityUri, bool validat
         internal static AuthorityInfo FromAadAuthority(Uri cloudInstanceUri, Guid tenantId, bool validateAuthority)
         {
 #pragma warning disable CA1305 // Specify IFormatProvider
-            return FromAuthorityUri(string.Format(CultureInfo.InvariantCulture, "{0}{1}/", cloudInstanceUri, tenantId.ToString("D")), validateAuthority);
+            return FromAuthorityUri(new Uri(cloudInstanceUri, tenantId.ToString("D")).ToString(), validateAuthority);
```

Comment:
Prefer using uri.AbsoluteUri instead of ToString(), see http://faithlife.codes/blog/2010/08/uritostring_must_die/